azurelinuxagent/agent.py

```diff
@@ -197,45 +217,58 @@ def parse_args(sys_args): # pylint: disable=R0912
     verbose = False
     debug = False
     conf_file_path = None
-    for a in sys_args: # pylint: disable=C0103
-        m = re.match("^(?:[-/]*)configuration-path:([\w/\.\-_]+)", a) # pylint: disable=W1401,C0103
+    log_collector_mode = None
+
+    for arg in sys_args: # pylint: disable=C0103
+        m = re.match("^(?:[-/]*)configuration-path:([\w/\.\-_]+)", arg) # pylint: disable=W1401,C0103
         if not m is None:
             conf_file_path = m.group(1)
             if not os.path.exists(conf_file_path):
                 print("Error: Configuration file {0} does not exist".format(
                         conf_file_path), file=sys.stderr) # pylint: disable=C0330
                 usage()
                 sys.exit(1)
-        
-        elif re.match("^([-/]*)deprovision\\+user", a):
+
+        m = re.match("^(?:[-/]*)-mode:([\w/\.\-_]+)", arg)
```

Comment:
The linter is complaining here because `\w` is not a valid escape. I think this should be a `r"..."` string.